elliott/elliottlib/cli/snapshot_cli.py

```diff
@@ -154,7 +154,16 @@ async def new_snapshot(self, build_records) -> dict:
         await self.konflux_client._get(API_VERSION, KIND_APPLICATION, application_name)
 
         async def _comp(record):
-            comp_name = KonfluxImageBuilder.get_component_name(application_name, record.name)
+            # Use stored component name from build record if available, otherwise generate default
+            if hasattr(record, 'component') and record.component:
+                comp_name = record.component
+            elif hasattr(record, 'bundle_component') and record.bundle_component:
```

Comment:
hmm i think these component types could be defined for a single image, making `elif` wrong.

You might also want to look into `Missing` handling elsewhere.